update: added a workaround on enabling IDE support in HMPP (#59)
JetBrains/kotlin-mobile-docs: topics/add-dependencies.md

            }
        }
    }
}
```
        
</tab>
<tab title="Kotlin">
    
```Kotlin
kotlin {
    sourceSets["commonMain"].dependencies {
        implementation(project(":some-other-multiplatform-module"))
    }
    sourceSets["androidMain"].dependencies {
        //platform part of :some-other-multiplatform-module will be added automatically
    }
}
```
</tab>
</tabs>

## iOS dependencies

Apple SDK dependencies (such as Foundation or Core Bluetooth) are available as a set of prebuilt libraries in a Kotlin Multiplatform Mobile project. 
They do not require any additional configuration.

You can also reuse other libraries and frameworks from the iOS ecosystem in your iOS source sets. 
Kotlin supports interoperability with Objective-C dependencies and Swift dependencies if their API is exported to Objective-C with the `@objc` attribute. 
Pure Swift dependencies are not yet supported.

Integration with the CocoaPods dependency manager is also supported with the same limitation – you cannot use pure Swift pods. 

We recommend:
* [Using CocoaPods](#with-cocoapods) to handle iOS dependencies in Kotlin Multiplatform Mobile (KMM) projects. 

> * [Managing dependencies manually](#without-cocoapods) only if you want to tune the interop process specifically or if you have some other strong reason to do so.

Reviewer: Are you sure about using the list here? Maybe it's just me but before reading the descriptions it looks like two **equally recommended** options, which is wrong.

Author: Yes, these are equally recommended options. And a bulleted list is used for a list of options. The list improves readability and adds the structure. People consume structured information easier than just text. And it's also a simple TOC of the section.
I don't get what's wrong with this list. Could you elaborate? :)

Reviewer: Maybe this wording confused me, but I see:
1. `handle iOS dependencies in Kotlin Multiplatform Mobile (KMM) projects` - which is pretty common. For non-expert like me this description looks like a general way.
2. **only** if you want to tune the interop process specifically or if you have **some other strong reason to do so**. - which seems to me an advanced way for expert users; it can be potentially dangerous (**strong reason to do so** is scary)

Author: OK. I see your point now. removed the list – kept 2 sentences instead :)

Learn more about [Objective-C and Swift interop](https://kotlinlang.org/docs/reference/native/objc_interop.html) and 
[configuring cinterop from Gradle](https://kotlinlang.org/docs/reference/mpp-dsl-reference.html#cinterops).

### Workaround to enable IDE support for the shared iOS source set {initial-collapse-state="collapsed"}

Due to the [known issue](https://youtrack.jetbrains.com/issue/KT-40975), you won't be able to use IDE features, such as code completion and  highlighting, for the shared iOS source set 
in a multiplatform project with [hierarchical structure support](https://kotlinlang.org/docs/reference/mpp-share-on-platforms.html#share-code-on-similar-platforms), if your project depends on:

* Multiplatform libraries that don't support the hierarchical structure.
* Third-party iOS libraries, except for [platform libraries](https://kotlinlang.org/docs/native-platform-libs.html) supported out of the box.

This issue applies only to the shared iOS source set. The IDE will correctly support all the other code.

> > All KMM projects created with the KMM Project Wizard support the hierarchical structure and can also have this issue.

Reviewer: ```suggestion
> All projects created with the KMM Project Wizard support the hierarchical structure and can also have this issue.
```

>
{type="note"}

> To enable IDE support in these cases, you can work around the issue and add the following code to `build.gradle.(kts)` in the `shared` directory of your project:

Reviewer: ```suggestion
To enable IDE support in these cases, you can work around the issue by adding the following code to `build.gradle.(kts)` in the `shared` directory of your project:
```

<tabs>

```Groovy
def iosTarget
if (System.getenv("SDK_NAME")?.startsWith("iphoneos")) {
    iosTarget = kotlin.&iosArm64
} else {
    iosTarget = kotlin.&iosX64
}
```

```Kotlin
val iosTarget: (String, KotlinNativeTarget.() -> Unit) -> KotlinNativeTarget =
    if (System.getenv("SDK_NAME")?.startsWith("iphoneos") == true)
        ::iosArm64
    else
        ::iosX64

iosTarget("ios")
```

</tabs>

> In this code sample, configuration of iOS targets depends on the environment variable `SDK_NAME`, which is managed by Xcode. 

Reviewer: Not sure about the meaning of this sentence. What is it for? Do you expect any actions or conclusions?

Reviewer 2: This is an explanation of how this workaround works, I add it in order to reduce the number of magic 🧙🏻‍♂️ users would copy to their build files

Author: Yes, it's a question from users. We've added this so that users could understand what the code does – not just blindly copy it.

Reviewer: Maybe explain _what_ this variable reflects, not just _where it comes from_? As an ios newbie, I don't feel that this sentence makes the hack much clearer. But I agree if you think that it's not for newbies.

Author: Let's keep this way. If we get questions from newbies, we'll provide more details. Don't want to overload for now.

Author: You can google in any case :) and find an answer.
